test/cpp/ext/otel/otel_test_library.h

```diff
@@ -54,31 +54,94 @@ class MockMetricReader : public opentelemetry::sdk::metrics::MetricReader {
   void OnInitialized() noexcept override {}
 };
 
-class OpenTelemetryPluginEnd2EndTest : public ::testing::Test {
- protected:
-  // Note that we can't use SetUp() here since we want to send in parameters.
-  void Init(
-      const absl::flat_hash_set<absl::string_view>& metric_names,
-      opentelemetry::sdk::resource::Resource resource =
-          opentelemetry::sdk::resource::Resource::Create({}),
-      std::unique_ptr<grpc::internal::LabelsInjector> labels_injector = nullptr,
-      bool test_no_meter_provider = false,
-      const std::map<std::string, std::string>& labels_to_inject = {},
-      absl::AnyInvocable<bool(absl::string_view /*target*/) const>
-          target_selector = absl::AnyInvocable<bool(absl::string_view) const>(),
+struct OpenTelemetryPluginTestConfiguration {
```

Comment:
Suggest moving this inside of `OpenTelemetryPluginEnd2EndTest` and calling it `Options`.

Reply:
done